### PR TITLE
#4730: sandbox nunjucks and handlebars calls from runtime

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Part of #4730
- Uses Chrome sandbox for calls to nunjucks and handlebars from runtime

## Demo

- No user-facing changes

## Remaining Work

- Update Nunjucks constructor call in variable analysis framework. Might need to shift work to sandbox?

## Checklist

- [x] Add tests
- [x] Run Storybook and manually confirm that all stories are working
- [x] Designate a primary reviewer: @fregante

### CODE REVIEW

File: src/sandbox/messenger/executor.ts [hunk 1]
```diff
@@ -15,16 +15,50 @@
...
 
-import { type ApplyJqPayload, type NunjucksRenderPayload } from "./api";
+import { type ApplyJqPayload, type TemplateRenderPayload } from "./api";
+import { once } from "lodash";
+import { isErrorObject } from "@/errors/errorHelpers";
+import { InvalidTemplateError } from "@/errors/businessErrors";
 
-export async function renderNunjucksTemplate(payload: NunjucksRenderPayload) {
-  const { template, context, autoescape } = payload;
+const ensureNunjucks = once(async () => {
```
Comment (fregante) [inline src/sandbox/messenger/executor.ts:23]: Once isn't needed yet, import() only executes the content once already

File: src/sandbox/messenger/executor.ts [hunk 2]
```diff
@@ -15,16 +15,50 @@
...
+  const { template, context, autoescape } = payload;
+  const nunjucks = await ensureNunjucks();
 
   nunjucks.configure({ autoescape });
-  return nunjucks.renderString(template, context);
+  try {
+    return nunjucks.renderString(template, context);
+  } catch (error) {
+    if (isErrorObject(error) && error.name === "Template render error") {
+      throw new InvalidTemplateError(error.message, template);
```
Comment (fregante) [inline src/sandbox/messenger/executor.ts:48]: Keep in mind that this error might not full serialize (definitely no "instanceof" at the moment). So I'd doublecheck this or just pass the plain Error as before instead; the wrapping could be done outside the sandbox.

If this is expected to be used in multiple places, it's good as is.

Comment (twschiller) [reply src/sandbox/messenger/executor.ts:37]: What's the concern? InvalidTemplateError only contains serializable properties: error message and template

> If this is expected to be used in multiple places, it's good as is.

The raw error from Nunjucks isn't very helpful. So I think it makes sense to just do the error transform here

Comment (fregante) [reply src/sandbox/messenger/executor.ts:37]: No particular concern, just a note that this no longer matches `instanceof InvalidTemplateError` _if_ we have that anywhere

File: webpack.config.js [hunk 3]
```diff
@@ -32,16 +32,26 @@ const mergeWithShared = require("./webpack.sharedConfig.js");
 
 function parseEnv(value) {
   switch (String(value).toLowerCase()) {
-    case "undefined":
+    case "undefined": {
```
Comment (twschiller) [inline webpack.config.js:35]: Lint rule

**Pull request merged**